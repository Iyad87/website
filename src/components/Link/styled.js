import styled from 'styled-components';
import { highlight } from '../../styles/colours';

export const StyledLink = styled.a`
  color: ${highlight};
  font-weight: 500;
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`;
