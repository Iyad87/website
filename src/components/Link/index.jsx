import GatsbyLink from 'gatsby-link';
import React from 'react';
import PropTypes from 'prop-types';
import { StyledLink } from './styled';

const internalRegExp = /^\/(?!\/)/;
export const isInternal = href => internalRegExp.test(href);

const InternalLink = StyledLink.withComponent(GatsbyLink);

const Link = ({ href, ...props }) => {
  if (isInternal(href)) return <InternalLink to={href} {...props} />;

  return <StyledLink href={href} {...props} />;
};

Link.propTypes = {
  href: PropTypes.string.isRequired,
};

export default Link;
