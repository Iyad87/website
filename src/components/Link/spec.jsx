import React from 'react';
import { shallow } from 'enzyme';
import Link, { isInternal } from './index';

describe('components/Link', () => {
  test('internal link snapshot', () => {
    const wrapper = shallow(<Link href="/test">Test</Link>);

    expect(wrapper).toMatchSnapshot();
  });

  test('external link snapshot', () => {
    const wrapper = shallow(<Link href="https://test.com/">Test</Link>);

    expect(wrapper).toMatchSnapshot();
  });

  test('uses GatsbyLink for internal links', () => {
    const wrapper = shallow(<Link href="/internal">Internal</Link>);

    expect(wrapper.is('Styled(GatsbyLink)')).toBe(true);
  });

  test('uses anchor for external links', () => {
    const wrapper = shallow(<Link href="https://external.com/">External</Link>);

    expect(wrapper.text()).toContain('styled.a');
  });

  test('isInternal', () => {
    expect(isInternal('http://google.com')).toBe(false);
    expect(isInternal('//external')).toBe(false);
    expect(isInternal('/test')).toBe(true);
    expect(isInternal('/')).toBe(true);
  });
});
