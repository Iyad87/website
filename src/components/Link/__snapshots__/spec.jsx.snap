// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Link external link snapshot 1`] = `
<styled.a
  href="https://test.com/"
>
  Test
</styled.a>
`;

exports[`components/Link internal link snapshot 1`] = `
<Styled(GatsbyLink)
  to="/test"
>
  Test
</Styled(GatsbyLink)>
`;
